rpc/legacyrpc/methods.go

```diff
@@ -3472,6 +3472,18 @@ func walletInfo(s *Server, icmd interface{}) (interface{}, error) {
 		}
 	}
 
+	coinType, err := w.CoinType()
+	if errors.Is(errors.WatchingOnly, err) {
+		// This is a watching-only wallet, which does not store the active coin
+		// type. Return CoinTypes default value (0), which will be omitted from
+		// the JSON response, and log a debug message.
+		log.Debug("Watching only wallets do not store the coin type keys.")
+	} else if err != nil {
+		log.Errorf("Failed to retrieve the active coin type: %v", err)
+		coinType = 0
+		// return with error or omit coin type?
```

Comment:
logging and omitting seems fine to me. the wallet probably never was able to sync if this error was hit otherwise, and any additional details we could still glean from RPCs may be valuable.